scripts/pythondistdeps.py

```diff
@@ -334,83 +366,71 @@ def normalize_name(name):
+                        dep_normalized_name = dep.name.lower().replace('_', '-')
 
                     if args.legacy:
-                        name = 'pythonegg({})({})'.format(pyver_major, dep.key)
+                        name = 'pythonegg({})({})'.format(pyver_major, dep.name.lower())
                     else:
                         if args.majorver_only:
                             name = 'python{}dist({})'.format(pyver_major, dep_normalized_name)
                         else:
                             name = 'python{}dist({})'.format(dist.py_version, dep_normalized_name)
-                    for spec in dep.specs:
-                        if name not in py_deps:
-                            py_deps[name] = []
-                        if spec not in py_deps[name]:
-                            py_deps[name].append(spec)
-                    if not dep.specs:
+                    if dep.marker and not args.recommends:
+                        if not dep.marker.evaluate(get_marker_env(dist, '')):
+                            continue
+                    if name not in py_deps:
                         py_deps[name] = []
+                    for spec in dep.specifier:
+                        if (spec.operator, spec.version) not in py_deps[name]:
+                            py_deps[name].append((spec.operator, spec.version))
             # Unused, for automatic sub-package generation based on 'extras' from egg/dist metadata
             # TODO: implement in rpm later, or...?
             if args.extras:
-                deps = dist.requires()
-                extras = dist.extras
-                print(extras)
-                for extra in extras:
+                print(dist.extras)
+                for extra in dist.extras:
                     print('%%package\textras-{}'.format(extra))
                     print('Summary:\t{} extra for {} python package'.format(extra, dist.key))
                     print('Group:\t\tDevelopment/Python')
-                    depsextras = dist.requires(extras=[extra])
-                    for dep in reversed(depsextras):
-                        if dep in deps:
-                            depsextras.remove(dep)
-                    deps = depsextras
-                    for dep in deps:
-                        for spec in dep.specs:
-                            if spec[0] == '!=':
-                                print('Conflicts:\t{} {} {}'.format(dep.key, '==', spec[1]))
+                    for dep in dist.requirements_for_extra(extra):
+                        for spec in dep.specifier:
+                            if spec.operator == '!=':
+                                print('Conflicts:\t{} {} {}'.format(dep.name, '==', spec.version))
                             else:
-                                print('Requires:\t{} {} {}'.format(dep.key, spec[0], spec[1]))
+                                print('Requires:\t{} {} {}'.format(dep.name, spec.operator, spec.version))
```

Comment:
Another two places where `dep.key` is being replaced by yet different expression. Maybe it would be worth creating the `key` property for `Distribution` and unifying it.

Reply:
@s-t-e-v-e-n-k Here you still use `dep.name` as a replacement for `dep.key`, is there a reason?